duckduckgo/Android
# Large scale https upgrades

Task/Issue URL: https://app.asana.com/0/72649045549333/721626625236432
Tech Design URL: https://app.asana.com/0/72649045549333/727813825116580

**Description**:
Add large scale https support to android
- Removes legacy https solution
- Adds updated https solution
- Adds new api requests 
- Clear old https data and perform database migration to support new format

**Steps to test this PR**:

TEST ONE:
1. Visit the http://wikipedia.org
1. Check the logs and ensure they report that this site is "upgradable"
1. Note that the scheme is updated to https

TEST TWO:
1. Visit the http://aaaaaaaaaaaaa.co.uk
1. Check the logs and ensure they report that this site is not "upgradable"
1. Note that the scheme remains as "http"
1. The site itself doesn't exist (thus definitely not in our upgrade list), so won't load

TEST THREE:
1. Install the app from develop and run it
1. Then install this version over the top
1. Ensure that the db migration succeeds and nothing goes boom!

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
